packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx

```diff
@@ -183,6 +208,39 @@ export function CategoryAutocomplete({
     [defaultCategoryGroups, categoryGroups, showSplitOption],
   );
 
+  const filterSuggestions = useMemo(
+    () =>
+      (
+        suggestions: (Suggestion | CategoryAutocompleteItem)[],
+        value: string,
+      ) => {
+        return suggestions
+          .filter((suggestion: Suggestion | CategoryAutocompleteItem) => {
+            const suggestionAsSuggestion = suggestion as Suggestion; // Cast to Suggestion
+            return (
+              suggestionAsSuggestion.id === 'split' ||
+              suggestionAsSuggestion.group?.name
+                .toLowerCase()
+                .includes(value.toLowerCase()) ||
+              (
+                suggestionAsSuggestion.group?.name +
+                ' ' +
+                suggestionAsSuggestion.name
+              )
+                .toLowerCase()
+                .includes(value.toLowerCase()) ||
+              defaultFilterSuggestion(suggestionAsSuggestion, value)
+            );
+          })
+          .sort(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏`sort` mutates the input (sorts in place). Lets use [`toSorted`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted) to do the same thing, but instead copy to a new object.